### PR TITLE
chore: unit test for duplicate message push

### PR DESCRIPTION
**Pull request by darshankabariya**
It's related to #2320 

It's test duplicate message push from filter service node to filter client

### CODE REVIEW

File: tests/wakunode_rest/test_rest_filter.nim [hunk 1]
```diff
@@ -315,3 +316,147 @@ suite "Waku v2 Rest API - Filter V2":
...
+    let messages1 = await restFilterTest.client.filterGetMessagesV1(DefaultContentTopic)
+
+    check:
+      postMsgResponse1.status == 200
+      $postMsgResponse1.contentType == $MIMETYPE_TEXT
+      postMsgResponse1.data == "OK"
+      len(messages1.data) == 1
+
+    # Pause execution for 2 minutes to test TimeCache functionality of service node
+    sleep(120000)
```
Comment (Ivansete-status) [inline tests/wakunode_rest/test_rest_filter.nim:447]: Oh, unfortunately we can't wait for so long in a single test

**Review by Ivansete-status — COMMENTED** [review]
Thanks for this! I haven't checked in in deep detail but we need to have fast tests :)
Cheers

**Review by NagyZoltanPeter — CHANGES_REQUESTED** [review]
Please adjust mountFilter with the dup check TTL config and than you are able to configure this test's timeout to be short.

**Comment by darshankabariya** [comment]
> Please adjust mountFilter with the dup check TTL config and than you are able to configure this test's timeout to be short.

I’ve been updating the tests based on your suggestions. However, I encountered an issue with await testSetup.serviceNode.mountFilter(cacheTTL = 1.seconds). I just need to override this individual parameter, but it's showing a type mismatch (one is TimeInterval and the other is Duration) i try many thing but issue is persistant. In my local setup, the tests are not running correctly, but I will still push the changes to the branch. Sometimes my local environment behaves differently from the remote setup. Let’s see what happens. If you find anything wrong, please let me know.

**Comment by NagyZoltanPeter** [comment]
> > Please adjust mountFilter with the dup check TTL config and than you are able to configure this test's timeout to be short.
> 
> I’ve been updating the tests based on your suggestions. However, I encountered an issue with `await testSetup.serviceNode.mountFilter(cacheTTL = 1.seconds).` I just need to override this individual parameter, but it's showing a type mismatch (one is TimeInterval and the other is Duration) i try many thing but issue is persistence. In my local setup, the tests are not running correctly, but I will still push the changes to the branch. Sometimes my local environment behaves differently from the remote setup. Let’s see what happens. If you find anything wrong, please let me know.

Yeah, saw and fixed that. It was a wrong import std/times instead of chronos/timer. Now it compiles I think.

**Review by NagyZoltanPeter — APPROVED** [review]
I think it will be ok now with 1 sec delay for test.

**Review by NagyZoltanPeter — CHANGES_REQUESTED** [review]
Compile error is fixed but test fails unfortunately:

```2024-06-28T00:09:14.7083858Z ================================================================
2024-06-28T00:09:14.7098245Z   /home/runner/work/nwaku/nwaku/build/all_tests_waku 'Waku v2 Rest API - Filter V2::duplicate message push to filter subscriber ( sleep in between )'
2024-06-28T00:09:14.7099205Z ----------------------------------------------------------------
2024-06-28T00:09:14.7099981Z     /home/runner/work/nwaku/nwaku/tests/wakunode_rest/test_rest_filter.nim(462, 26): Check failed: len(messages2.data) == 1
2024-06-28T00:09:14.7100699Z     len(messages2.data) was 0
2024-06-28T00:09:14.7100903Z 
2024-06-28T00:09:14.7101289Z   [FAILED ] (  0.02s) duplicate message push to filter subscriber ( sleep in between )[0m
2024-06-28T00:09:14.7101714Z 
2024-06-28T00:09:14.7101870Z (0.09s)   Waku v2 Rest API - Filter V2```

File: tests/wakunode_rest/test_rest_filter.nim [hunk 2]
```diff
@@ -315,3 +317,147 @@ suite "Waku v2 Rest API - Filter V2":
...
+    let messages1 = await restFilterTest.client.filterGetMessagesV1(DefaultContentTopic)
+
+    check:
+      postMsgResponse1.status == 200
+      $postMsgResponse1.contentType == $MIMETYPE_TEXT
+      postMsgResponse1.data == "OK"
+      len(messages1.data) == 1
+
+    # Pause execution for 2 minutes to test TimeCache functionality of service node
+    sleep(1000)
```
Comment (NagyZoltanPeter) [inline tests/wakunode_rest/test_rest_filter.nim:448]: I see, it was the issue!
Maybe to change it for a more commonly used version:
```suggestion
    await sleepAsync(1.seconds)
```

File: tests/wakunode_rest/test_rest_filter.nim [hunk 3]
```diff
@@ -315,3 +317,147 @@ suite "Waku v2 Rest API - Filter V2":
...
+    # check messages received client side or not
+    let messages1 = await restFilterTest.client.filterGetMessagesV1(DefaultContentTopic)
+
+    check:
+      postMsgResponse1.status == 200
+      $postMsgResponse1.contentType == $MIMETYPE_TEXT
+      postMsgResponse1.data == "OK"
+      len(messages1.data) == 1
+
+    # Pause execution for 2 minutes to test TimeCache functionality of service node
```
Comment (NagyZoltanPeter) [inline tests/wakunode_rest/test_rest_filter.nim:447]:
```suggestion
    # Pause execution for 1 second to test TimeCache functionality of service node
```

**Review by NagyZoltanPeter — APPROVED** [review]
As soon as the tests pass it is very well done!!!

File: tests/wakunode_rest/test_rest_filter.nim (hunk 2 above)
Comment (NagyZoltanPeter) [reply tests/wakunode_rest/test_rest_filter.nim:448]: Just a side note: using async sleep will not stop the whole thread so async tests can run meanwhile.

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks! Make sure Zoltán's comments are considered before merging :)

**Comment by darshankabariya** [comment]
> LGTM! Thanks! Make sure Zoltán's comments are considered before merging :)

Sure !

File: tests/wakunode_rest/test_rest_filter.nim (hunk 2 above)
Comment (darshankabariya) [reply tests/wakunode_rest/test_rest_filter.nim:448]: > await sleepAsync(1.seconds)

oh that's awesome !

**Pull request merged**